src/test/regress/expected/shard_rebalancer_unit.out

```diff
@@ -108,7 +108,7 @@ SELECT unnest(shard_placement_rebalance_array(
           '{"shardid":4, "nodename":"hostname2"}'
         ]::json[]
 ));
-WARNING:  Not allowed to move shard xxxxx anywhere from hostname2:5432
+WARNING:  There is not any node to move shard xxxxx from hostname2:5432
```

Comment:
I think this sentence is missing a word, but I'm also not sure if this new wording is the best. Maybe something like this is better:
> Shard xxx needs to be moved away from hostname2:5432, but there are no nodes it can be moved to.

Or 

> Shard xxx needs to be moved away from hostname2:5432, but there are no nodes it is allowed to be moved to.

The reason why it said "Allowed" in the original message is that this doesn't only happen when there's no nodes at all. It also happens when there are nodes, but these nodes are not allowed to contain this node, such as nodes with should_have_shards=true or nodes not matching the shard_allowed_on_node function.